src/Workspaces/Core/Portable/CodeRefactorings/FixAllOccurences/FixAllState.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.FixAll;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings
+{
+    internal partial class FixAllState
+    {
+        internal readonly int CorrelationId = LogAggregator.GetNextId();
+
+        public FixAllProvider FixAllProvider { get; }
+        public CodeAction CodeAction { get; }
+        public CodeRefactoringProvider CodeRefactoringProvider { get; }
+        public Document? Document { get; }
+        public Project Project { get; }
+        public FixAllScope FixAllScope { get; }
+        public Solution Solution => this.Project.Solution;
```

Comment:
it is possible for FixAllState to subclass a CommonFixAllState for both fixes and refactorings?